File: programmers/JS_language/동영상 재생기.js
function solution(video_len, pos, op_start, op_end, commands) {
  //prev 10초 전 이동
  //next 10초 후 이동
  //동영상 길이 video_len, 재생위치 pos,
  //시작시간 op_start, 종료시간 op_end

  const [vmin, vsec] = video_len.split(":").map(Number);
  let vTime = vmin * 60 + vsec;

  const [min, sec] = pos.split(":").map(Number);

  let realTime = min * 60 + sec;

  const [start_min, start_sec] = op_start.split(":").map(Number);
  const [end_min, end_sec] = op_end.split(":").map(Number);

  const realStart = start_min * 60 + start_sec;
  const realEnd = end_min * 60 + end_sec;

  //오프닝 사이인 경우
  if (realTime >= realStart && realTime <= realEnd) {
    realTime = realEnd;
  }

  for (let i of commands) {
    // prev
    if (i === "prev") realTime = Math.max(0, realTime - 10);
    // next
    if (i === "next") realTime = Math.min(vTime, realTime + 10);

    //오프닝 사이인 경우 재확인인
    if (realTime >= realStart && realTime <= realEnd) {
      realTime = realEnd;
    }
  }

  let changeMin = Math.floor(realTime / 60);
  let changeSec = realTime % 60;

  // 두 자리 문자열로 포맷 (선택)
  changeMin = String(changeMin).padStart(2, "0");
  changeSec = String(changeSec).padStart(2, "0");

  return [changeMin, changeSec].join(":");
}

File: programmers/C_language/동영상 재생기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(
    const char* video_len,
    const char* pos,
    const char* op_start,
    const char* op_end,
    const char* commands[],
    size_t commands_len
) {

    int vmin, vsec;
    sscanf(video_len, "%d:%d", &vmin, &vsec);

    int vTime = vmin * 60 + vsec;

    int min, sec;
    sscanf(pos, "%d:%d", &min, &sec);

    int realTime = min * 60 + sec;

    int start_min, start_sec;
    sscanf(op_start, "%d:%d", &start_min, &start_sec);

    int end_min, end_sec;
    sscanf(op_end, "%d:%d", &end_min, &end_sec);

    int realStart = start_min * 60 + start_sec;
    int realEnd = end_min * 60 + end_sec;

    //오프닝 사이인 경우
    if(realTime >= realStart && realTime <= realEnd){
        realTime = realEnd;
    }

    for(int i = 0; i < commands_len; i++){

        // prev
        if(strcmp(commands[i], "prev") == 0){

            realTime -= 10;

            if(realTime < 0)
                realTime = 0;
        }

        // next
        else if(strcmp(commands[i], "next") == 0){

            realTime += 10;

            if(realTime > vTime)
                realTime = vTime;
        }

        //오프닝 사이인 경우 재확인
        if(realTime >= realStart && realTime <= realEnd){
            realTime = realEnd;
        }
    }

    int changeMin = realTime / 60;
    int changeSec = realTime % 60;

    // 결과 문자열 메모리 할당
    char* answer = (char*)malloc(6);

    sprintf(answer, "%02d:%02d", changeMin, changeSec);

    return answer;
}